### PR TITLE
Bugfix for text registers k, y, d, o, c, t, and p not working.

### PR DESCRIPTION
**Pull request by justinhopkins**
The previous configuration was triggering *Rectangle commands because they were unaware of emacs-mcx.inRegisterSaveMode or emacs-mcx.inRegisterInsertMode being active.

### CODE REVIEW

**Comment by whitphx** [comment]
Thank you for finding the bug and contributing the fix!

I think, however, setting the `acceptingRectCommand` context as `false` at the same time [when `inRegisterSaveMode` is set as `true`](https://github.com/whitphx/vscode-emacs-mcx/blob/main/src/commands/registers.ts#L12) would be better, because the `when` settings become simpler and it looks semantically correct in that these contexts are exclusive.

If you took care of the synchronization of the `acceptingRectCommand` and the `StartAcceptingRectCommand.acceptingRectCommand` flag here, thank you for the care but you don't have to. The `acceptingRectCommand` flag was introduced only to avoid frequent calls of `executeCommand()` for performance. (Sorry the code here is not descriptive enough)
https://github.com/whitphx/vscode-emacs-mcx/blob/44561f05894a8e34160b06a468b3597800a7af5a/src/commands/rectangle.ts#L29-L32

**Comment by justinhopkins** [comment]
Good point! Will push changes when I have them ready

**Comment by justinhopkins** [comment]
Changes pushed. Please have a look. I think if this implementation works (I've tested it), it is much, much cleaner.

File: keybindings.json [hunk 1]
```diff
@@ -793,7 +793,7 @@
     {
       "key": "t", // ctrl+x r t
       "command": "emacs-mcx.stringRectangle",
-      "when": "emacs-mcx.acceptingRectCommand && editorTextFocus"
+      "when": "emacs-mcx.acceptingRectCommand && editorTextFocus)"
```
Comment (whitphx) [inline keybindings.json]:
```suggestion
      "when": "emacs-mcx.acceptingRectCommand && editorTextFocus"
```

**Review by whitphx — COMMENTED** [review]
LGTM!

Just dropped a single comment, thanks!

**Comment by whitphx** [comment]
Thanks!

**Pull request merged**